### PR TITLE
B051

### PR DESCRIPTION
**Pull request by juzen2003**
# Update:
* Display a modal when cart status api call has error returned. Call loadData to render the page again.
* Rename tableSorting to reRenderData

### CODE REVIEW

File: opus/application/apps/ui/templates/base.html [hunk 1]
```diff
@@ -289,18 +289,31 @@ <h3 class='op-header-text'></h3>
...
             id_name="op-empty-cart"
             confirm_title="Confirm Remove All Observations From Cart"
             confirm_msg="Are you sure you want to remove all observations from your cart?"
             target="op-empty-cart"
+            yes_text="Yes"
+            no_text="No"
+        %}
+        {% include "confirm_modal.html" with
+            id_name="op-cart-status-error-msg"
+            confirm_title="Message About Cart Status"
```
Comment (rfrenchseti) [inline opus/application/apps/ui/templates/base.html]: It's only possible to have a failure return with a status code on add/addrange/addall, so let's make the title something more meaningful to the user: "Failed to Add Observations to Cart".

File: opus/application/apps/ui/templates/confirm_modal.html [hunk 2]
```diff
@@ -12,8 +12,12 @@
         {{confirm_msg}}
       </div>
       <div class="modal-footer">
-        <button type="submit" class="btn btn-primary" data-dismiss="modal" data-target={{target}}>Yes</button>
-        <button type="cancel" class="btn btn-danger">No</button>
+        {% if yes_text != None %}
```
Comment (rfrenchseti) [inline opus/application/apps/ui/templates/confirm_modal.html]: You can just say `{% if yes_text %}` and `{% if no_text %}`

File: opus/application/static_media/js/cart.js [hunk 3]
```diff
@@ -388,6 +388,19 @@ var o_cart = {
 
         o_cart.lastCartRequestNo++;
         $.getJSON(url  + add_to_url + "&reqno=" + o_cart.lastCartRequestNo, function(statusData) {
+            if (statusData.reqno < o_cart.lastCartRequestNo) {
```
Comment (rfrenchseti) [inline opus/application/static_media/js/cart.js]: Ooooo...this is going to be a problem. While we only want to display the cart badge number for the final API call, any one of the intermediate calls might return an error that the user cares about. So you have to pop up the dialog for *all* returns, but only update the badge for the "last" one. Just move this if below the statusData if below and it should be fine.

**Review by rfrenchseti — CHANGES_REQUESTED** [review]
Now that I see the return error messages, they are pretty ugly to the user. Once Dave finishes this change I'll update the error messages in the backend.

**Comment by juzen2003** [comment]
# Update: 
* B051
  - Update modal title to "Failed to Add Observations to Cart".
  - Disable the ability to close modal when clicking outside of modal area
  - Update to have statusDataErrorCollector and display the whole series of error modals one after another. New modal will show up when previous one is close.

**Review by rfrenchseti — APPROVED** [review]

**Review by dstopp — APPROVED** [review]

File: opus/application/settings.py [hunk 4]
```diff
@@ -352,7 +352,7 @@
 
 THUMBNAIL_NOT_FOUND = 'https://tools.pds-rings.seti.org/static_media/img/thumbnail_not_found.png'
 
-MAX_SELECTIONS_ALLOWED = 10000
+MAX_SELECTIONS_ALLOWED = 3
```
Comment (dstopp) [inline opus/application/settings.py]: oops probably you don't want to check this in :)

File: opus/application/static_media/css/opus.css [hunk 5]
```diff
@@ -236,8 +236,9 @@ ul.searchMenu {
     opacity: 0.15;
 }
 
-#confirm-modal {
+#op-confirm-modal {
     border-radius: 6px;
+    border: solid 3px #D3D3D3;
```
Comment (dstopp) [inline opus/application/static_media/css/opus.css]: please use a bootstrap color and not a hard-coded color.

**Pull request merged**